common.sh

```diff
@@ -160,9 +175,247 @@ exit_fixed()
     exit $RESULT_FIXED
 }
 
+_get_cached_command()
+{
+    # Function gets a cached command
+    # input parameter is command provided by get_cached function
+    command=$1
+    if [ ! -f "$command" ]; then
+        log_error "File $command does not exist. It is mandatory."
+    else
```

Comment:
I do not want to repeat test for existing file or command more then once.
See what you proposed:

    get_cached() {
        #
        # Get general system information from cache.
        #
        # Usage:  get_cached file <PATH>
        #         get_cached command <KEY>
        #         get_cached filelist <KEY>
        #         get_cached info <KEY>
        #
        # First sub-command is used to get cached version of file
        # that was originally on <PATH>.  Following paths are
        # currently cached:
        #
        #  *  /etc/passwd
        #  *  /etc/group
        #
        # Second sub-command `command` retrieves cached output of
        # a common system interrogation command identified by `<KEY>`,
        # which can be one of:
        #
        #  *  rpm_qa - output of `rpm -qa`
        #  *  rpm_Va - output of `rpm -Va`
        #
        # Third sub-command `filelist` gets cached file list named
        # <KEY>.  Format is one path per line and currently provided
        # lists are:
        #
        #  *  executable - (describe meaning)
        #  *  allmyfiles - (describe meaning)
        #
        # Last sub-command `info` provides access to miscellaneous
        # information cached by Preupgrade Assistant.  Valid keys
        # are:
        #
        # *  foo - (describe meaning and format)
        # *  bar - (describe meaning and format)
        #
        true
    }